File: deep-vision-1.1/src/ImageLoaderThread.java
import java.util.concurrent.BlockingQueue;

public class ImageLoaderThread implements Runnable {

	private final BlockingQueue<ImageData> processQueue;
	private static final String IMAGE_PATH = "D:\\Vision_Application\\hul_bad_230219\\Bad";
	private static final String EXT = ".BMP";

	public ImageLoaderThread(BlockingQueue processQueue) {
		this.processQueue = processQueue;
	}

	@Override
	public void run() {
		try {
			loadImage();
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
	}

	private void loadImage() throws InterruptedException {
		for (int i = 11; i <= 60; i++) {
			ImageData image = new ImageData();
			image.setImgName("Image001" + i + EXT);
			image.setPath(IMAGE_PATH);
			
			Thread.sleep(100);
			this.processQueue.put(image);

		}
	}

}

File: deep-vision-py/com/deepvision/simulator/ImageLoaderThread.py
import threading
import time
from com.deepvision.constants import Constant
from com.deepvision.models.ImageData import ImageData
import cv2 as cv2


class ImageLoaderThread(threading.Thread):

    def __init__(self, name, imageProcessQueue):
        threading.Thread.__init__(self)
        self.name = name
        self.imageProcessQueue = imageProcessQueue

    def run(self):
        self.loadImages()

    def loadImages(self):
        for i in range(7):
            # img = ImageData()
            # img.image_name = 'Image001{}.{}'.format(i, ".BMP")
            # img.image_path = Constant.IMAGE_PATH
            # img.image = cv2.imread(img.image_path + "\\" + img.image_name, 0)

            # self.imageProcessQueue.put(Constant.IMAGE_PATH+"\\"+'Image001{}{}'.format(i+14, ".BMP"))
            # self.imageProcessQueue.put("D:\\Vision_Application\\OCR\\OCR 9\\Image004{}{}".format(77+i, ".BMP"))   # job 14
            self.imageProcessQueue.put("D:\\Vision_Application\\ffc\\" + 'Image000{}{}'.format(i+12, ".BMP"))   # job 15
            # print('loading images... {}'.format(Constant.IMAGE_PATH+"\\"+'Image001{}{}'.format(i+14, ".BMP")))
            # time.sleep(.1)
